src/Kupo/Data/Cardano/Address.hs

```diff
@@ -47,8 +48,9 @@ unsafeAddressFromBytes =
 
 addressFromBytes :: ByteString -> Maybe Address
 addressFromBytes bytes =
-    Ledger.deserialiseAddr bytes
+    Ledger.Api.decodeAddr bytes
   <|>
+    -- TODO: This is probably redundant (now with decodeAddr and/or decodeAddrLenient
```

Comment:
We must use `decodeAddrLenient` here. `decodeAddr` is just broken and doesn't work on historical data :)